### PR TITLE
Add option to change app icon for better visibility

### PR DESCRIPTION
**Pull request by Takuro-Ito**
## Changes
- Use app icon as status bar icon
  - Now it's easier to recognize MonitorControl in menu bar

### Before
image

### After
image

#### Menu Icon Style: Sun (Default)
image

#### Menu Icon Style: App Icon (New)
image

### CODE REVIEW

File: MonitorControl/Support/AppDelegate.swift [hunk 1]
```diff
@@ -71,11 +71,6 @@ class AppDelegate: NSObject, NSApplicationDelegate {
     }
     prefs.set(currentBuildNumber, forKey: PrefKey.buildNumber.rawValue)
     self.setDefaultPrefs()
-    if !DEBUG_MACOS10, #available(macOS 11.0, *) {
-      self.statusItem.button?.image = NSImage(systemSymbolName: "sun.max", accessibilityDescription: "MonitorControl")
-    } else {
-      self.statusItem.button?.image = NSImage(named: "status")
-    }
```
Comment (Takuro-Ito) [inline MonitorControl/Support/AppDelegate.swift:78]: Moved to `MenuHandler.swift` to update Menu Bar Icon when user changes in preferences panel.

**Review by JoniVR — CHANGES_REQUESTED** [review]
image

The spacing between the menu items seems off.

Also, I'm not sure how I feel about this change, since it doesn't follow HID,  and adds an option of which I'm sure people will want more custom icons in the future. I'm just not sure if it's worth adding imo.

[Relevant HID](https://developer.apple.com/design/human-interface-guidelines/macos/extensions/menu-bar-extras/):
image


See #748 for similar discussion. 

@waydabber @the0neyouseek thoughts?

**Comment by Takuro-Ito** [comment]
Thanks for reviewing and adding missing localized strings!
I will fix spacing later today. 

About menu bar icon, the motivation to this PR is that I've had trouble with finding MonitorControl from many menu bar icons. 
Although the sun icon conforms to HID, it's hard to distinguish custom functionality which is Monitor Control app from macOS provided functionality. 
Using colored icon as menu bar doesn't conform HID but it's very very easy to find. 

At least if the menu bar icon is unique and represents MonitorControl well, I would be much appreciated (so I don't have to merge to my fork every new release).

**Comment by waydabber** [comment]
Well, I agree with @JoniVR on this one, but I know its a very subjective topic. As mentioned on the other Issue about the icon I'd add a silent option in the preferences file which would allow tinkerers to change the icon string (SFString) and also we could add a similar option to use the app icon as menu icon. I am not sure about adding it as an option in preferences though.

**Comment by Takuro-Ito** [comment]
Just the fixed layout issue.

Yes, a silent option with the app icon would be enough for me without adding an option in preferences!

**Comment by waydabber** [comment]
@JoniVR - what should be done about this PR? We should make a decision, it should not sit here unresolved I think. :) I say we should go to the silent option route (I'll add a preferences option which can be edited with a plist editor that allows changing the icon to an other SFIcon - since this issue comes up from time to time, users who has a need for this can be directed to a Q&A entry about it). I think making this an exposed option is not a good idea, breaks consistency, the app will look in all kinds of (maybe bad) ways on screenshots etc.

What do you say?

**Comment by the0neyouseek** [comment]
@waydabber I think it's the best option here

**Comment by waydabber** [comment]
Ok, let's agree on this solution then. I'll open a new Feature Request about it. I am working on the v4.0.1 at the moment but this will solely focus on bugs and some internal improvements, I won't add any new features, but in the next feature release we'll add this one.

https://github.com/MonitorControl/MonitorControl/issues/766

I'll close this PR for now then.

Many thanks @Takuro-Ito for the understanding and the contribution regarding this!

**Comment by JoniVR** [comment]
I think the problem (and solution) of this PR was mostly visibility (Color icon), which afaik won't be resolved by simply adding a customised SFIcon string. 

But I guess we can't make everyone happy and I honestly don't feel like we should implement a custom icon picker either. So the SFIcon string might indeed be a good alternative for people that just want a custom icon.

**Comment by waydabber** [comment]
well, we can further elaborate the options by making this an option either and also allow multicolored icon rendering (introduced in Monterey). But maybe that's too much for a silent option like this.

**Comment by kevintoepfer** [comment]
I'd like to add a suggestion from an UX standpoint: MonitorControl should have an own icon that better represents "controlling the monitor" instead of "controlling brightness of whatever". That icon should not break HID. Maybe a combination of a monitor + sun symbol would help.

What do you think about it, @waydabber @JoniVR? I would be happy to assist in creating an icon if you are interested.

**Comment by waydabber** [comment]
Well, changing the icon is a controvelsian one. By all means, you can create a proposal, but it is not guaranteed that it will be accepted + an unanimous decision is needed for a change among the dev team. :)

**Pull request closed without merging**